Homework #1. Task #1. Волынец Арсений Викторович

vol0n opened the pull request.

vol0n committed:
Inital commit

vol0n committed:
Fix detect errors, enable autocorrection in detect plugin

nbirillo commented on src/main/kotlin/homework_1/task_1.kt:
```diff
@@ -0,0 +1,30 @@
+fun factorialRec(n: Int): Int = if (n <= 1) 1 else n * factorialRec(n - 1)
```
А где имя пакета?

nbirillo replied:
А почему ты не используешь ее? Считай итеративной и рекурсивной версиями

nbirillo commented on build.gradle.kts:
```diff
@@ -21,7 +21,8 @@ dependencies {
 
 detekt {
-    failFast = true // fail build on any finding
+    failFast = false // fail build on any finding
```
А зачем ты это поменял?)

vol0n replied:
у меня было такое, что детект выдаёт ошибки я их пытался исправить, а ничего не происходило, поэтому я что-то менял и обратно забыл поставить failFast, исправлю)

nbirillo commented on src/main/kotlin/homework_1/task_1.kt:
```diff
@@ -0,0 +1,30 @@
+fun factorialRec(n: Int): Int = if (n <= 1) 1 else n * factorialRec(n - 1)
+
+fun factorialIter(n: Int): Int {
+    var res = n
```
Для 0 неверно сработает

nbirillo commented on src/main/kotlin/homework_1/task_1.kt:
```diff
@@ -0,0 +1,30 @@
+    return res
+}
+
+val scan = java.util.Scanner(System.`in`)
```
Давай обойдемся без глобальных переменных

vol0n committed:
Fix factorialIter func

Description: directory renamed, added package name, corrected factorialIter func, removed global var

vol0n committed:
Change detect PackageNaming option

nbirillo reviewed (approved):
Зачтено!
